### PR TITLE
Move polyfill JavaScript code to be included as a remote script from …

### PR DESCRIPTION
**Pull request by jonathanKingston**
### Description
Inline `<script` tags are a source of code injection, the browser can protect against this with CSP to block inline scripts. However Magento has a lot of inline scripts that need removing bit-by-bit.

### Fixed Issues (if relevant)
1. Removal of an inline script block

### Manual testing scenarios
1. Ensure browsers with no localstorage still support storage

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by jonathanKingston** [comment]
Updating to fix the coding style since it was moved...

**Comment by jonathanKingston** [comment]
Updated code style which is now passing locally.

**Review by VladimirZaets — APPROVED** [review]

File: app/code/Magento/Theme/view/frontend/layout/default_head_blocks.xml [hunk 1]
```diff
@@ -10,10 +10,10 @@
         <meta name="viewport" content="width=device-width, initial-scale=1"/>
         <css src="mage/calendar.css"/>
         <script src="requirejs/require.js"/>
+        <script src="mage/polyfill.js"/>
     </head>
     <body>
         <referenceContainer name="after.body.start">
-            <block class="Magento\Framework\View\Element\Template" name="head.polyfill" as="polyfill" template="Magento_Theme::js/polyfill.phtml" before="-"/>
```
Comment (VladimirZaets) [inline app/code/Magento/Theme/view/frontend/layout/default_head_blocks.xml]: Hi @jonathanKingston. Can you please fix it uses the backward compatible way. We can't delete block from the layout. Please look our document for more details about backward compatibility. You can leave the block declaration, and use `<script type="text/x-magento-init">` to initialize js file.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @jonathanKingston. Please fix it using the backward compatible way. We can't delete block from the layout. Please look at our document for more details about backward compatibility. You can leave the block declaration, and use <script type="text/x-magento-init"> to initialize js file.

**Comment by jonathanKingston** [comment]
@VladimirZaets I'm a little confused by your comment, this should be fully backwards compatible no? If I use an init block it will be asynchronously loaded where as this will load synchronously in the same manner as the embedded block currently.
Are you suggesting that the `app/code/Magento/Theme/view/frontend/templates/js/polyfill.phtml` file should still exist? If so the only way would be for this to import the new JS file I have created.

**Comment by VladimirZaets** [comment]
@jonathanKingston We had discussion about this case and take desition that your fix is ok, so we will merge your fix in the near time, thanks.

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @jonathanKingston. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.3.0 release.

**Pull request merged**